torch/testing/_internal/opinfo/core.py

```diff
@@ -731,6 +731,20 @@ class OpInfo:
     dtypes: _dispatch_dtypes = None
 
     # the following dtypesIf... options override the dtypes value on their respective device types
+    dtypesIf: dict[str, _dispatch_dtypes] = field(default_factory=dict)
+
+    def __getattribute__(self, name: str) -> Any:
+        if name.startswith("dtypesIf") and name != "dtypesIf":
+            dev_name = name[len("dtypesIf") :].lower()
+            return self.dtypesIf.get(dev_name)
+        return super().__getattribute__(name)
+
+    def __setattr__(self, name: str, value: Any) -> None:
```

Comment:
Use a TypeVar for the Value so we can have proper type checking of the value against the values in the dict

Reply:
Perhaps I miss something, but value is really `Any` as setattr will be called for all attributes of the class.
But added an assert when set the values for `dtypesIf`